File: Emulator/src/main/content/global/skill/summoning/familiar/npc/SpiritPengatriceNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.summoning.familiar.Familiar;
import content.global.skill.summoning.familiar.FamiliarSpecial;
import core.game.node.entity.player.Player;
import org.rs.consts.Items;
import org.rs.consts.NPCs;

/**
 * The type Spirit pengatrice npc.
 */
public class SpiritPengatriceNPC extends content.global.skill.summoning.familiar.Familiar {

    /**
     * Instantiates a new Spirit pengatrice npc.
     */
    public SpiritPengatriceNPC() {
        this(null, NPCs.SPIRIT_PENGATRICE_6883);
    }

    /**
     * Instantiates a new Spirit pengatrice npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public SpiritPengatriceNPC(Player owner, int id) {
        super(owner, id, 3600, Items.SP_PENGATRICE_POUCH_12103, 3);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new SpiritPengatriceNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        return false;
    }

    @Override
    public int[] getIds() {
        return new int[]{NPCs.SPIRIT_PENGATRICE_6883, NPCs.SPIRIT_PENGATRICE_6884};
    }

}

File: Server/src/main/content/global/skill/summoning/familiar/npc/SpiritPengatriceNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.summoning.familiar.Familiar
import content.global.skill.summoning.familiar.FamiliarSpecial
import core.game.node.entity.player.Player
import shared.consts.Items
import shared.consts.NPCs

class SpiritPengatriceNPC
@JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.SPIRIT_PENGATRICE_6883) :
    Familiar(owner, id, 3600, Items.SP_PENGATRICE_POUCH_12103, 3) {
    override fun construct(owner: Player, id: Int): Familiar {
        return SpiritPengatriceNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        return false
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.SPIRIT_PENGATRICE_6883, NPCs.SPIRIT_PENGATRICE_6884)
    }
}